keras_cv/models/object_detection/retina_net/__internal__/layers/prediction_head.py

```diff
@@ -27,30 +29,42 @@ def PredictionHead(output_filters, bias_initializer):
       A function representing either the classification
         or the box regression head depending on `output_filters`.
     """
-    conv_layers = [
-        layers.Conv2D(
-            256,
-            3,
-            padding="same",
-            kernel_initializer=initializers.RandomNormal(0.0, 0.01),
-            activation="relu",
-        )
-        for _ in range(4)
-    ]
-    conv_layers += [
-        layers.Conv2D(
-            output_filters,
-            3,
-            1,
-            padding="same",
-            kernel_initializer=initializers.RandomNormal(0.0, 0.01),
-            bias_initializer=bias_initializer,
-        )
-    ]
-
-    def apply(x):
-        for layer in conv_layers:
-            x = layer(x)
+
+    def __init__(self, output_filters, bias_initializer, **kwargs):
+        super().__init__(**kwargs)
+        self.output_filters = output_filters
+        self.bias_initializer = bias_initializer
+        conv_layers = [
+            layers.Conv2D(
+                256,
+                3,
+                padding="same",
+                kernel_initializer=initializers.RandomNormal(0.0, 0.01),
```

Comment:
not sure why the kernel initializer is hardcoded here. Might need some comments here.

Reply:
Added a comment